### PR TITLE
Dev data processing

### PR DESCRIPTION
**Pull request by ehutt**
First pull request. Initial prototype using subset of case document data. Loads and cleans data, creates baseline LDA model, has some visualizations for the LDA model. 

Please check if you are able to run it on your machines and don't be afraid to leave comments! I am trying to learn better coding practices, so please give feedback.

### CODE REVIEW

**Comment by bghill** [comment]
It's great to see you doing your updates as a PR. I'll go through this later this morning.

**Review by bghill — APPROVED** [review]
Please add: nltk and gensim to the requirements.txt before merging.
Normally, I would add a number of small code cleanup items.  But given the timeline, I'll stick to the science.  Overall this looks good.  Clearly the word clouds look suspicious, but that is just a side-effect of the size of the test-sample.

File: source/dataClean.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Jun 11 14:50:34 2019
+
+@author: elizabethhutton
+"""
+
+
+#Notes:
```
Comment (RohanBhandari) [inline source/dataClean.py]: Remove notes from code. You can keep them in a separate, uncommited file.

File: source/dataClean.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+#to stem or not to stem? 
+
+import pickle
+import pandas as pd
+import nltk
+from nltk.tokenize import word_tokenize
+from nltk.corpus import stopwords
+from nltk.stem.porter import PorterStemmer
+
+#download packages if first time running
```
Comment (RohanBhandari) [inline source/dataClean.py]: Move these instructions to the setup section of your README

File: source/dataClean.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import nltk
+from nltk.tokenize import word_tokenize
+from nltk.corpus import stopwords
+from nltk.stem.porter import PorterStemmer
+
+#download packages if first time running 
+#nltk.download('punkt')
+#nltk.download('stopwords')
+
+#function: dataClean
```
Comment (RohanBhandari) [inline source/dataClean.py:20]: Nice, clear documentation. Look into "docstrings", which is the python convention for documentation.

File: source/main.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+#starting with only AK cases from 2008
+api_AK = 'https://api.case.law/v1/cases/?jurisdiction=ark&decision_date_min=2008-12-31&full_case=true'
+raw_file = 'data/documents_raw.pkl'
+clean_file = 'data/documents_clean.pkl' 
+
+#process data
+dataDownload(api_AK,raw_file)
+dataClean(raw_file,clean_file,STEM=True)
+
+#define vars for LDA
```
Comment (RohanBhandari) [inline source/main.py:28]: Add argument parsing, so variables can be set from command line. See the `argparse` module.

**Review by RohanBhandari — APPROVED** [review]
Hi Elizabeth,

Nice work on your project so far! It's great that you have a model up and running. I made a few comments, but there is one larger code execution issue. 

When you import a file in python, it runs all the lines in the file. This means that when you import functions from `dataIngest` and `dataClean` these scripts get run, including [these lines](https://github.com/ehutt/text2topics/blob/3fc08373a6e7aa79521ffc48fe401e1e44dd55b8/source/dataIngest.py#L54-L57) and [these lines](https://github.com/ehutt/text2topics/blob/3fc08373a6e7aa79521ffc48fe401e1e44dd55b8/source/dataClean.py#L58-L60). So you actually end up running `dataDownload` and `dataClean` twice, which is why data files get created in the `source/` dir and the `source/data/` dir. The way to prevent this is to place an if statement above those lines: `if __name__ == __main__:`. This if statement will not evaluate True when the file is imported and the lines won't run. I highly suggest you make that change before merging.

Overall, though, great job, and I look forward to seeing the progress you make this week!

**Pull request merged**